src/main/java/bio/terra/stairway/Database.java

```diff
@@ -260,19 +257,22 @@ public void step(FlightContext flightContext) {
             return;
         }
 
+        final String sqlInsertFlightLog =
+                "INSERT INTO " + FLIGHT_LOG_TABLE +
+                        "(flightid, log_time, working_parameters, step_index, doing, succeeded, error_message)" +
+                        " VALUES (?, CURRENT_TIMESTAMP, ?, ?, ?, ?, ?)";
+
         try (Connection connection = dataSource.getConnection();
-             Statement statement = connection.createStatement()) {
+             PreparedStatement statement = connection.prepareStatement(sqlInsertFlightLog)) {
 
-            statement.executeUpdate(
-                    "INSERT INTO " + FLIGHT_LOG_TABLE +
-                            "(flightid, log_time, working_parameters, step_index, doing, succeeded, error_message)" +
-                            " VALUES ('" +
-                            flightContext.getFlightId() + "', CURRENT_TIMESTAMP, '" +
-                            flightContext.getWorkingMap().toJson() + "'," +
-                            flightContext.getStepIndex() + "," +
-                            boolString(flightContext.isDoing()) + "," +
-                            boolString(flightContext.getResult().isSuccess()) + ", '" +
-                            flightContext.getResult().getErrorMessage() + "')");
+            statement.setString(1, flightContext.getFlightId());
```

Comment:
I prefer the pattern of using named parameters in the SQL string instead of using positional indexes like this. It will make it easier to when reading this code or making updates later. Surprisingly JDBC doesn't look like it supports this, but it looks like we can use Spring's [NamedParameterJdbcTemplate](https://docs.spring.io/spring/docs/2.5.x/javadoc-api/org/springframework/jdbc/core/namedparam/NamedParameterJdbcTemplate.html). 

I think this is close, just requires swapping some question marks with `:name`s and replacing indexes with `"name"`s.

Reply:
The stairway package doesn't depend on Spring. It doesn't use JdbcTemplate. My impulse is to keep it that way. There are several standalone named parameter wrappers that I could include instead.

Reply:
Okay that's fair. I'd prefer if we could stick close to the JdbcTemplate approach since that is how I'd like to build SQL on the repository metadata side.

Reply:
Sounds good to me. I'll make it so.